sql/sys_vars.cc

```diff
@@ -3826,6 +3826,15 @@ static bool fix_rpl_semi_sync_master_wait_point(sys_var *self, THD *thd,
   return false;
 }
 
+static bool fix_rpl_semi_sync_master_wait_for_slave_count
+  (sys_var *self, THD *thd, enum_var_type type)
+{
+  mysql_mutex_unlock(&LOCK_global_system_variables);
+    repl_semisync_master.refresh_wait_for_slave_count(thd->variables.server_id);
```

Comment:
Ideally, the refresh is only required when the requirement decreases.
However, it seems that the Sys.Var. system has a limitation that the variable must have an ‘old’ duplicate that an `ON_UPDATE` callback can compare to tell whether it was decreased.

Intriguingly, other Semi-Sync Primary variables _are_ duplicated and updated through `ON_UPDATE` even though they don’t have special handling.
This is likely a remnant of Semi-Sync’s past as a plugin.